src/test/java/org/mule/tools/apikit/AbstractScaffolderTestCase.java

```diff
@@ -92,115 +96,41 @@ public static MuleDomain createMuleDomainFromLocation(String location) throws Ex
-                                          List<MuleConfig> existingMuleConfigs) {
-    return scaffoldApi(runtimeEdition, apiReference, null, existingMuleConfigs, true);
-  }
-
-  protected ScaffoldingResult scaffoldApi(RuntimeEdition runtimeEdition, ApiReference apiReference, MuleDomain muleDomain,
-                                          List<String> existingMuleConfigsLocations)
-      throws Exception {
-    List<MuleConfig> existingMuleConfigs = createMuleConfigsFromLocations(existingMuleConfigsLocations);
-    return scaffoldApi(runtimeEdition, apiReference, muleDomain, existingMuleConfigs, true);
-  }
-
-  protected ScaffoldingResult scaffoldApi(RuntimeEdition runtimeEdition, ApiReference apiReference, MuleDomain muleDomain,
-                                          List<MuleConfig> existingMuleConfigs, List<String> existingMuleConfigsLocations)
-      throws Exception {
-    List<MuleConfig> existingMuleConfigsFromLocation = createMuleConfigsFromLocations(existingMuleConfigsLocations);
-    existingMuleConfigsFromLocation.addAll(existingMuleConfigs);
-    return scaffoldApi(runtimeEdition, apiReference, muleDomain, existingMuleConfigsFromLocation, true);
-  }
-
-  protected ScaffoldingResult scaffoldApi(RuntimeEdition runtimeEdition, String ramlLocation,
-                                          MuleDomain muleDomain, List<MuleConfig> existingMuleConfigs) {
-    return scaffoldApi(runtimeEdition, ramlLocation, muleDomain, existingMuleConfigs, true);
-  }
-
-  protected ScaffoldingResult scaffoldApi(RuntimeEdition runtimeEdition, String ramlLocation,
-                                          MuleDomain muleDomain, List<MuleConfig> existingMuleConfigs, boolean showConsole) {
-    ApiReference apiReference = ApiReference.create(Paths.get(ramlLocation).toString());
-    return scaffoldApi(runtimeEdition, apiReference, muleDomain, existingMuleConfigs, showConsole);
-  }
-
-  protected ScaffoldingResult scaffoldApi(RuntimeEdition runtimeEdition, ApiReference apiReference,
-                                          MuleDomain muleDomain, List<MuleConfig> existingMuleConfigs, boolean showConsole) {
+                                          MuleDomain muleDomain, List<MuleConfig> existingMuleConfigs,
+                                          ScaffoldingAccessories scaffoldingAccessories) {
     ScaffolderContext context = ScaffolderContextBuilder.builder().withRuntimeEdition(runtimeEdition).build();
     MainAppScaffolder mainAppScaffolder = new MainAppScaffolder(context);
-
+    if (scaffoldingAccessories == null) {
+      scaffoldingAccessories = new ScaffoldingAccessories(true, null, null, null);
+    }
     ScaffoldingConfiguration scaffoldingConfiguration =
-        getScaffoldingConfiguration(apiReference, existingMuleConfigs, muleDomain, showConsole);
+        getScaffoldingConfiguration(apiReference, existingMuleConfigs, muleDomain, scaffoldingAccessories);
```

Comment:
There are a lot of tests invoking `scaffoldApi` with `scaffoldingAccessories` and a bunch of additional arguments being `null`. I think we should pass `ScaffoldingConfiguration` object as an argument from the caller and that's it. We would avoid all the parameters of this method and this should only instantiate the MainAppScaffolder and run it.